### PR TITLE
Updates the Issues API to use the new pattern

### PR DESCRIPTION
**Pull request by dhruvsinghal**
@pengwynn please take a look.

### CODE REVIEW

**Comment by pengwynn** [comment]
Awesome, @dhruvsinghal. Mind tackling https://github.com/octokit/go-octokit/issues/72 as part of this PR?

**Comment by dhruvsinghal** [comment]
@pengwynn #72 changes are in.

File: octokit/issues.go [hunk 1]
```diff
@@ -1,49 +1,80 @@
...
-func (i *IssuesService) All() (issues []Issue, result *Result) {
-	result = i.client.get(i.URL, &issues)
+func (i *IssuesService) All(uri *Hyperlink, uriParams M) (issues []Issue,
+	result *Result) {
+	if uri == nil {
+		uri = &RepoIssuesURL
+	}
+	url, err := uri.Expand(uriParams)
+	if err != nil {
+		return make([]Issue, 0), &Result{Err: err}
```
Comment (pengwynn) [inline octokit/issues.go]: Any reason to return an empty array here rather than `nil`?

File: octokit/issues.go [hunk 2]
```diff
@@ -1,49 +1,80 @@
...
-
 	"github.com/jingweno/go-sawyer/hypermedia"
+	"time"
 )
 
 // RepoIssuesURL is a template for accessing issues in a particular
 // repository for a particular owner that can be expanded to a full address.
-var RepoIssuesURL = Hyperlink("repos/{owner}/{repo}/issues{/number}")
+var RepoIssuesURL = Hyperlink("repos/{owner}/{repo}/issues{/number}?" +
+	"filter={filter}&state={state}&labels={labels}&sort={sort}&" +
```
Comment (pengwynn) [inline octokit/issues.go]: I think this should be:

```go
Hyperlink("repos/{owner}/{repo}/issues{/number}{?filter,state,labels,sort}")
```

[RFC 6750][rfc] makes for some great bedtime reading. :night_with_stars: 

[rfc]: https://tools.ietf.org/html/rfc6570

File: octokit/issues.go (hunk 1 above)
Comment (dhruvsinghal) [inline octokit/issues.go]: My bad: I didn't realize that you could return `nil` in place of slices.

File: octokit/issues.go (hunk 2 above)
Comment (dhruvsinghal) [inline octokit/issues.go]: I'll try reading it that way...so far I've only tried reading an OS textbook at bedtime!

**Comment by dhruvsinghal** [comment]
@pengwynn Please take a look.

File: octokit/issues.go (hunk 2 above)
Comment (feiranchen) [inline octokit/issues.go]: @dhruvsinghal I think @pengwynn meant for you to update the line (the regex) to what he commented haha

**Comment by feiranchen** [comment]
@dhruvsinghal ^note on the comment above

File: octokit/issues.go (hunk 2 above)
Comment (dhruvsinghal) [inline octokit/issues.go]: Sorry, I thought I made the changed, but apparently I didn't.

**Comment by dhruvsinghal** [comment]
@pengwynn Please take yet another look.

**Comment by pengwynn** [comment]
@dhruvsinghal Looks good. We'll merge after we figure out what's up with CI.

**Pull request merged**